Snowflake.Data/Logger/SecretDetector.cs

```diff
@@ -0,0 +1,166 @@
+﻿/*
+ * Copyright (c) 2021 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Snowflake.Data.Log
+{
+    class SecretDetector
+    {
+        public struct Mask
+        {
+            public Mask(bool isMasked = false, string maskedText = null, string errStr = null)
+            {
+                this.isMasked = isMasked;
+                this.maskedText = maskedText;
+                this.errStr = errStr;
+            }
+
+            public bool isMasked { get; set; }
+            public string maskedText { get; set; }
+            public string errStr { get; set; }
+        }
+
+        private static List<string> CUSTOM_PATTERNS_REGEX = new List<string>();
+        private static List<string> CUSTOM_PATTERNS_MASK = new List<string>();
+        private static int CUSTOM_PATTERNS_LENGTH;
+
+        public static void SetCustomPatterns(string[] customRegex, string[] customMask)
+        {
+            if (CUSTOM_PATTERNS_LENGTH != 0)
+            {
+                ClearCustomPatterns();
+            }
+
+            if (customRegex.Length == customMask.Length)
+            {
+                CUSTOM_PATTERNS_LENGTH = customRegex.Length;
+                for (int index = 0; index < CUSTOM_PATTERNS_LENGTH; index++)
+                {
+                    CUSTOM_PATTERNS_REGEX.Add(customRegex[index]);
+                    CUSTOM_PATTERNS_MASK.Add(customMask[index]);
+                }
+            }
+            else
+            {
+                throw new ArgumentException("Regex count and mask count must be equal.");
+            }
+        }
+
+        public static void ClearCustomPatterns()
+        {
+            CUSTOM_PATTERNS_REGEX.Clear();
+            CUSTOM_PATTERNS_MASK.Clear();
+            CUSTOM_PATTERNS_LENGTH = 0;
+        }
+
+        private static string MaskCustomPatterns(string text)
+        {
+            string result;
+            for (int index = 0; index < CUSTOM_PATTERNS_LENGTH; index++)
+            {
+                result = Regex.Replace(text, CUSTOM_PATTERNS_REGEX[index], CUSTOM_PATTERNS_MASK[index],
+                                         RegexOptions.IgnoreCase);
+
+                if (result != text)
+                {
+                    return result;
+                }
+            }
+            return text;
+        }
+
+        private static readonly string AWS_KEY_PATTERN = @"(aws_key_id|aws_secret_key|access_key_id|secret_access_key)\s*=\s*'([^']+)'";
```

Comment:
Can we update this to include 

`'AWS_SECRET_KEY': 'bajbpafjb;a;adjpboajbvjavbjadj'`

I think the current pattern only accounts for `=` signs and not `:`, also quotes around the key

Reply:
Ok